spec/system/admin/reports/sales_tax/sales_tax_totals_by_order_spec.rb

```diff
@@ -425,56 +428,63 @@
+          it "downloads the file" do
+            select output_type, from: "report_format"
+
+            expect { generate_report }.to change { downloaded_filenames.length }.from(0).to(1)
+
+            expect(downloaded_filename).to match(/.*\.#{extension}/)
+
+            downloaded_file_txt = load_file_txt(extension, downloaded_filename)
+
+            expect(downloaded_file_txt).to include "DistributorEnterpriseWithTax", "oc1",
+                                                   "ORDER_NUMBER_1", "GST Food"
+            expect(downloaded_file_txt).to include "State", "cfname", "clname", "ABC123",
+                                                   "order1@example.com"
+            # order1, first tax rate line
+            expect(downloaded_file_txt).to match /115.0\s+1.73\s+116.73/
+            # order1, second tax rate line
+            expect(downloaded_file_txt).to match /115.0\s+2.88\s+117.88/
+            # order2, first tax rate line
+            expect(downloaded_file_txt).to match /215.0\s+3.23\s+218.23/
+            # order2, second tax rate line
+            expect(downloaded_file_txt).to match /215.0\s+5.38\s+220.38/
+
+            expectation = expect_totals ? :to : :not_to
+            # correct totals with white-space between them
+            expect(downloaded_file_txt).public_send(expectation,
+                                                    match(/TOTAL\s+215.0\s+8.61\s+223.61/))
+            # correct totals with white-space between them
+            expect(downloaded_file_txt).public_send(expectation,
+                                                    match(/TOTAL\s+115.0\s+4.61\s+119.61/))
+          end
         end
       end
 
-      context "xlsx files" do
-        let(:report_file_xlsx) do
-          File.open("spec/fixtures/reports/sales_tax_by_order/sales_tax_by_order.xlsx")
-        end
-
-        it 'downloads the file' do
-          expect(downloaded_filenames.length).to eq(0) # downloads folder should be empty
-          select "Spreadsheet", from: "report_format"
-          find("#display_summary_row").uncheck
-          click_on "Go"
-          wait_for_download
-          expect(downloaded_filenames.length).to eq(1) # downloads folder should contain 1 file
-          expect(downloaded_filename).to match(/.*\.xlsx/)
-          downloaded_content = extract_xlsx_rows(downloaded_filename, 1..5)
-          fixture_content = extract_xlsx_rows(report_file_xlsx, 1..5)
-          expect(downloaded_content).to eq(fixture_content)
-        end
+      it_behaves_like "reports generated as", "CSV", "csv", false
+      it_behaves_like "reports generated as", "Spreadsheet", "xlsx", true
+      it_behaves_like "reports generated as", "PDF", "pdf", true
+    end
+  end
+end
 
-        def extract_xlsx_rows(file, range)
-          xlsx = Roo::Excelx.new(file)
-          range.map { |i| xlsx.row(i) }
-        end
-      end
+def generate_report
+  click_on "Go"
+  wait_for_download
+end
```

Comment:
The indentation is still wrong, I'll fix this up and squash everything.

Reply:
whoops, I didn't think to look. Thanks for that.

Reply:
Ah thank you! I was not aware this would go within the `describe` block. But I noticed the spec worked as well, even if outside of it. This is probably very basic Rails knowledge I'm lacking here :see_no_evil: 

I see this is in Ready to Go, so I'll merge now. Thanks for all the help and input.